src/Components/MasterComponnets/Navbar/Navbar.tsx

```diff
@@ -0,0 +1,152 @@
+import MoreIcon from "@mui/icons-material/MoreVert";
+import NotificationsIcon from "@mui/icons-material/Notifications";
+import SearchIcon from "@mui/icons-material/Search";
+import { Avatar } from "@mui/material";
+import AppBar from "@mui/material/AppBar";
+import Badge from "@mui/material/Badge";
+import Box from "@mui/material/Box";
+import IconButton from "@mui/material/IconButton";
+import Menu from "@mui/material/Menu";
+import MenuItem from "@mui/material/MenuItem";
+import Toolbar from "@mui/material/Toolbar";
+import { Search, SearchIconWrapper, StyledInputBase } from "./NavbarStyle";
+import { useContext, useState } from "react";
+import { AuthContext } from "../../../Context/AuthContext/AuthContext";
+
+export default function Navbar() {
+  const { userData } = useContext(AuthContext);
+
+  // State to manage the anchor element for the mobile menu
+  const [mobileMoreAnchorEl, setMobileMoreAnchorEl] =
+    useState<null | HTMLElement>(null);
+
+  const isMobileMenuOpen = Boolean(mobileMoreAnchorEl);
+
+  const handleMobileMenuClose = () => {
+    setMobileMoreAnchorEl(null);
+  };
+
+  const handleMobileMenuOpen = (event: React.MouseEvent<HTMLElement>) => {
+    setMobileMoreAnchorEl(event.currentTarget);
+  };
+
+  const menuId = "primary-search-account-menu";
+
+  const mobileMenuId = "primary-search-account-menu-mobile";
+  // JSX to render the mobile menu
+  const renderMobileMenu = (
+    <Menu
+      anchorEl={mobileMoreAnchorEl}
+      anchorOrigin={{
+        vertical: "top",
+        horizontal: "right",
+      }}
+      id={mobileMenuId}
+      keepMounted
+      transformOrigin={{
+        vertical: "top",
+        horizontal: "right",
+      }}
+      open={isMobileMenuOpen}
+      onClose={handleMobileMenuClose}
+    >
+      {/* --------------- Menu item for notifications --------------- */}
+      <MenuItem>
+        <IconButton
+          size="large"
+          aria-label="show 5 new notifications"
+          color="inherit"
+        >
+          <Badge badgeContent={5} color="error">
+            <NotificationsIcon />
+          </Badge>
+        </IconButton>
+        <p>Notifications</p>
+      </MenuItem>
+
+      {/* --------------- Menu item for profile --------------- */}
+      <MenuItem>
+        <IconButton
+          size="large"
+          aria-label="account of current user"
+          aria-controls="primary-search-account-menu"
+          aria-haspopup="true"
+          color="inherit"
+        >
+          <Avatar sx={{ backgroundColor: "#3252DF" }}>
+            {userData.UserName.slice(0, 1).toUpperCase()}
```

Comment:
https://mui.com/material-ui/react-avatar/#fallbacks
![image](https://github.com/user-attachments/assets/ff43272f-6861-451f-a9d5-a2e9ab05debe)

If we provided an alt text with user name it will fallback automatically to the first letter if there is no image.